sql/analyzer/optimize_joins.go

```diff
@@ -351,18 +332,27 @@ func extractJoinColumnExpr(e sql.Expression) (leftCol *columnExpr, rightCol *col
 			return nil, nil
 		}
 
-		leftCol, ok := left.(*expression.GetField)
-		if !ok {
-			return  nil, nil
-		}
-
-		rightCol, ok := right.(*expression.GetField)
-		if !ok {
+		leftField, rightField := extractGetField(left), extractGetField(right)
+		if leftField == nil || rightField == nil {
 			return nil, nil
 		}
 
-		return &columnExpr{leftCol, right, e}, &columnExpr{rightCol, left, e}
+		leftCol = &columnExpr{leftField, left, right, e}
+		rightCol = &columnExpr{rightField, right, left, e}
+		return leftCol, rightCol
 	default:
 		return nil, nil
 	}
 }
+
+func extractGetField(e sql.Expression) *expression.GetField {
+	var field *expression.GetField
+	sql.Inspect(e, func(expr sql.Expression) bool {
```

Comment:
As discussed, make this thing bail if there's more than one table referenced in this sql.Expression. Not sure about mutli-field access on the same table...

Reply:
Fixed. Multi-field access from the same table silently succeeds, which is correct since the GetField expression is never actually used, but kind of confusing (colExpr type assumes a single GetField, didn't want to change at this point)